core/src/main/java/org/elasticsearch/action/admin/indices/recovery/TransportRecoveryAction.java

```diff
@@ -151,13 +125,18 @@ protected ClusterBlockException checkRequestBlock(ClusterState state, RecoveryRe
         return state.blocks().indicesBlockedException(ClusterBlockLevel.READ, concreteIndices);
     }
 
-    static class ShardRecoveryRequest extends BroadcastShardRequest {
+    static class ShardRecoveryRequest extends BaseNodeBroadcastRequest<RecoveryRequest> {
```

Comment:
I wonder if we should just in line the response as well? we don't expose it through the API anymore. Also, can we call this NodeRecoveryRequest?